example/ida/utils.py

```diff
@@ -66,61 +67,85 @@ def guess_machine():
-                                           idaapi.SCOLOR_RPTCMT))
-    elif isinstance(expr, m2_expr.ExprCompose):
-        s = "{"
-        s += ", ".join(["%s, %s, %s" % (expr2colorstr(regs_ids, subexpr),
+            return out
+
+class ExprSlice_ida(ExprSlice):
+    def __str__(self):
+        s = "%s[%s:%s]" % (self.arg.str_sub_expr(),
+                           idaapi.COLSTR(str(self.start),
+                                         idaapi.SCOLOR_RPTCMT),
+                           idaapi.COLSTR(str(self.stop),
+                                         idaapi.SCOLOR_RPTCMT))
+        return s
+
+class ExprMem_ida(ExprMem):
+    def __str__(self):
+        return '%s[%s]' % (idaapi.COLSTR('@' + str(self.size),
+                                         idaapi.SCOLOR_RPTCMT),
+                           self.arg)
+
+class ExprCond_ida(ExprCond):
+    pass
+
+class ExprCompose_ida(ExprCompose):
+    def __str__(self):
+        s = '{'
+        s += ", ".join(["%s, %s, %s" % (subexpr,
                                         idaapi.COLSTR(str(idx),
                                                       idaapi.SCOLOR_RPTCMT),
                                         idaapi.COLSTR(str(idx + subexpr.size),
                                                       idaapi.SCOLOR_RPTCMT))
-                        for idx, subexpr in expr.iter_args()])
-        s += "}"
-    else:
-        s = str(expr)
+                        for idx, subexpr in self.iter_args()])
+        s += '}'
+        return s
+
+class ExprOp_ida(ExprOp):
+    pass
+
+class ExprAff_ida(ExprAff):
+    pass
+
+def expr2exprida(regs_ids, expr):
+    if expr.is_int():
+        new_expr = ExprInt_ida(expr.arg, expr.size)
+    elif expr.is_id():
+        new_expr =  ExprId_ida(expr.name, expr.size)
+    elif expr.is_slice():
+        new_expr =  ExprSlice_ida(expr2exprida(regs_ids, expr.arg), expr.start, expr.stop)
+    elif expr.is_mem():
+        new_expr =  ExprMem_ida(expr2exprida(regs_ids, expr.arg), expr.size)
+    elif expr.is_cond():
+        new_expr =  ExprCond_ida(expr2exprida(regs_ids, expr.cond),
+                                 expr2exprida(regs_ids, expr.src1),
+                                 expr2exprida(regs_ids, expr.src2))
+    elif expr.is_compose():
+        new_expr =  ExprCompose_ida(*[expr2exprida(regs_ids, arg) for arg in expr.args])
+    elif expr.is_op():
+        new_expr =  ExprOp_ida(expr.op, *[expr2exprida(regs_ids, arg) for arg in expr.args])
+    elif expr.is_aff():
+        new_expr =  ExprMem_ida(expr2exprida(regs_ids, expr.dst),
+                                expr2exprida(regs_ids, expr.src))
+    new_expr.regs_ids = regs_ids
+    return new_expr
 
-    return s
+
+def expr2colorstr(regs_ids, expr):
+    """Colorize an Expr instance for IDA
+    @regs_ids: list of ExprId corresponding to available registers
+    @expr: Expr instance to colorize
+    """
+    return expr2exprida(regs_ids, expr)
```

Comment:
I will keep the ExprIda generation and the color generation in two different function (I have updated the code)